R/deltaccd.R

```diff
@@ -152,33 +152,46 @@ calcCCD = function(
     stop('refCor must be a correlation matrix, with identical rownames and colnames.')}
 
   geneNames = rownames(refCor)[rownames(refCor) %in% rownames(emat)]
-  refNow = refCor[geneNames, geneNames]
-  if (length(geneNames) < 2) {
-    stop('Fewer than two genes in the reference are in the expression matrix.')
-  } else if (length(geneNames) < nrow(refCor)) {
-    warning(sprintf('%d gene(s) in reference is/are not in the expression matrix.',
-                    nrow(refCor) - length(geneNames)))}
+  if (length(geneNames) < nrow(refCor)) {
+    stop(sprintf('%d gene(s) in reference is/are not in the expression matrix.',
+                 nrow(refCor) - length(geneNames)))}
 
   if (is.null(groupVec)) {
     groupVec = rep('all', ncol(emat))
   } else if (length(groupVec) != ncol(emat)) {
     stop('Length of groupVec does not match the number of columns in emat.')
   } else if (min(table(groupVec)) < 3) {
     stop('Each unique group in groupVec must have at least three samples.')}
+  
+  varCheck = foreach (groupNow = sort(unique(groupVec)), .combine = rbind) %do% {
```

Comment:
Duplicated code is a good indication of the need for a function.

Reply:
There's still duplicated code in the `if` and `stop`. Where should that code go?

Reply:
It probably would make sense to put it in the `checkVar` function.